tests/test_prompts.py

```diff
@@ -135,7 +158,6 @@ def test_prompt_kwargs():
     def test_kwarg_tpl(var, other_var="other"):
         """{{var}} and {{other_var}}"""
 
-    assert test_kwarg_tpl.template == "{{var}} and {{other_var}}"
```

Comment:
Why not keep these two `.template` checks?

Reply:
Because `template` is no longer a `str` but a `jinja2.template`, I could try adapting them instead of suppressing them, however, they seemed a bit redundant to me, WDYT?

Reply:
Yeah, probably type checks will be enough, though I would suggest adding assertions to verify `.signature`: check it to be present from decorator call `from_str` and None otherwise.

Reply:
I added a few more `assert` statements to check that the `signature`. It's not `None` only when the `Prompt` is constructed using the decorator _(otherwise, it can't infer the signature argument since there is no function)_.